### ISSUE TITLE
I have to install TF 1.8!

### ISSUE THREAD

**Issue by FB43** [tensorflow/tensorflow#37094]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows (10 or 8.1)
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):
- TensorFlow version:1.8
- Python version:3.6
- Installed using virtualenv? pip? conda?:both
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:9.1
- GPU model and memory:NVIDIA GeForce GTX 760

**Describe the problem**
Hello, the computing capabilities of my GPU is 3.0 and TF doesn't work in this case. I've read that it's possible to run the 1.8 version here https://medium.com/@naarkie/using-tensorflow-gpu-on-a-compute-3-0-graphics-card-in-windows-4184f4228fe9
But impossible to find the good package.
I need it to use the Visions of Chaos software (fractal art).
May somebody help me?
Thanks for all.
Best regards,
François
**Provide the exact sequence of commands / steps that you executed before running into the problem**

**Any other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by ravikyram** [comment]
@FB43 

Can you go through the[ link ](https://www.deciphertechnic.com/install-tensorflow-gpu-18-on-windows/)and see if it helps you.Also, see tested builded configuration from [here](https://www.tensorflow.org/install/source_windows#gpu).Thanks!

**Comment by FB43** [comment]
@ravikyram thanks, but in the link, there is: "pip install --ignore-installed --upgrade tensorflow-gpupip install --ignore-installed --upgrade tensorflow-gpu". And, of course, it will install the last tensoflow version, not the 1.8 which seems not to be available on the repository. That's my problem! If I can find the 1.8 package (source or build), it will be fine but it seems to have disappeared!
Any idea to find it?
Thanks for your help,
François

**Comment by FB43** [comment]
@ravikyram hello, I'm going on but I'm stuck there, any idea to continue?
Thanks, François
PS C:\windows\system32\tensorflow> ./bazel build --config=v1 //tensorflow/tools/pip_package:build_pip_package
Extracting Bazel installation...
.............
WARNING: Config values are not defined in any .rc file: v1
ERROR: in target '//external:cc_toolchain': no such package '@local_config_cc//': Traceback (most recent call last):
        File "C:/users/fbour/appdata/local/temp/_bazel_fbour/nseddbsr/external/bazel_tools/tools/cpp/cc_configure.bzl", line 37
                configure_windows_toolchain(repository_ctx)
        File "C:/users/fbour/appdata/local/temp/_bazel_fbour/nseddbsr/external/bazel_tools/tools/cpp/windows_cc_configure.bzl", line 323, in configure_windows_toolchain
                _find_env_vars(repository_ctx, vc_path)
        File "C:/users/fbour/appdata/local/temp/_bazel_fbour/nseddbsr/external/bazel_tools/tools/cpp/windows_cc_configure.bzl", line 194, in _find_env_vars
                _find_vcvarsall_bat_script(repository_ctx, vc_path)
        File "C:/users/fbour/appdata/local/temp/_bazel_fbour/nseddbsr/external/bazel_tools/tools/cpp/windows_cc_configure.bzl", line 188, in _find_vcvarsall_bat_script
                auto_configure_fail((vcvarsall + " doesn't exist, pl..."))
        File "C:/users/fbour/appdata/local/temp/_bazel_fbour/nseddbsr/external/bazel_tools/tools/cpp/lib_cc_configure.bzl", line 30, in auto_configure_fail
                fail(("\n%sAuto-Configuration Error:%...)))

Auto-Configuration Error: C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\VC\VCVARSALL.BAT doesn't exist, please check your VC++ installation
INFO: Elapsed time: 21,381s
FAILED: Build did NOT complete successfully (3 packages loaded)

**Comment by FB43** [comment]
@ravikyram VC++ seems OK. Now, it's Python and I Don't know why!!!
PYTHON_BIN_PATH is configured and the path variable too.
Thanks for your help.
PS C:\WINDOWS\system32\tensorflow> bazel build --config=v1 //tensorflow/tools/pip_package:build_pip_package
.
WARNING: Config values are not defined in any .rc file: v1
WARNING: C:/users/fbour/appdata/local/temp/_bazel_fbour/nseddbsr/external/protobuf_archive/WORKSPACE:1: Workspace name in C:/users/fbour/appdata/local/temp/_bazel_fbour/nseddbsr/external/protobuf_archive/WORKSPACE (@com_google_protobuf) does not match the name given in the repository's definition (@protobuf_archive); this will cause a build error in future versions
ERROR: C:/windows/system32/tensorflow/third_party/py/numpy/BUILD:11:1: no such package '@local_config_python//': Traceback (most recent call last):
        File "C:/windows/system32/tensorflow/third_party/py/python_configure.bzl", line 291
                _create_local_python_repository(repository_ctx)
        File "C:/windows/system32/tensorflow/third_party/py/python_configure.bzl", line 254, in _create_local_python_repository
                _get_python_include(repository_ctx, python_bin)
        File "C:/windows/system32/tensorflow/third_party/py/python_configure.bzl", line 210, in _get_python_include
                _execute(repository_ctx, [python_bin, "-c",...())"], <2 more arguments>)
        File "C:/windows/system32/tensorflow/third_party/py/python_configure.bzl", line 54, in _execute
                _fail("\n".join([error_msg.strip() if ... ""]))
        File "C:/windows/system32/tensorflow/third_party/py/python_configure.bzl", line 27, in _fail
                fail(("%sPython Configuration Error:%...)))
Python Configuration Error: Problem getting python include path.
java.io.IOException: ERROR: src/main/native/windows/processes-jni.cc(239): CreateProcessW("C:\Python27" -c "from __future__ import print_function;from distutils import sysconfig;print(sysconfig.get_python_inc())"): AccÞs refusÚ.
Is the Python binary path set up right? (See ./configure or PYTHON_BIN_PATH.) Is distutils installed?
 and referenced by '//third_party/py/numpy:headers'
ERROR: Analysis of target '//tensorflow/tools/pip_package:build_pip_package' failed; build aborted: Loading failed
INFO: Elapsed time: 77,898s
FAILED: Build did NOT complete successfully (79 packages loaded)
    currently loading: tensorflow/core ... (11 packages)
    Fetching https://mirror.bazel.build/github.com/LMDB/lmdb/archive/LMDB_0.9.19.tar.gz
    Fetching https://mirror.bazel.build/github.com/jemalloc/jemalloc/archive/4.4.0.tar.gz
    Fetching https://mirror.bazel.build/github.com/google/re2/archive/26cd968b735e227361c9703683266f01e5df7857.tar.gz
    Fetching https://mirror.bazel.build/github.com/google/snappy/archive/1.1.7.tar.gz
    Fetching https://mirror.bazel.build/.../abseil/abseil-cpp/archive/720c017e30339fd1786ce4aac68bc8559736e53f.tar.gz
    Fetching https://mirror.bazel.build/www.kurims.kyoto-u.ac.jp/~ooura/fft.tgz; 6,980b
    Fetching https://github.com/google/highwayhash/archive/dfcb97ca4fe9277bf9dc1802dd979b071896453b.tar.gz; 21,098b
    Fetching https://github.com/google/snappy/archive/1.1.7.tar.gz

**Comment by FB43** [comment]
@ravikyram I'm trying other thing in the link you sent me:
bazel build --config=opt --config=cuda --define=no_tensorflow_py_deps=true //tensorflow/tools/pip_package:build_pip_package
Maybe that's better but it doesn't works too I've checked, the file cudnn.h is really there!
Thanks for your help,
François
PS C:\WINDOWS\system32\tensorflow> bazel build --config=opt --config=cuda --define=no_tensorflow_py_deps=true //tensorflow/tools/pip_package:build_pip_package
ERROR: Skipping '//tensorflow/tools/pip_package:build_pip_package': error loading package 'tensorflow/tools/pip_package': Encountered error while reading extension file 'cuda/build_defs.bzl': no such package '@local_config_cuda//cuda': Traceback (most recent call last):
        File "C:/windows/system32/tensorflow/third_party/gpus/cuda_configure.bzl", line 1142
                _create_local_cuda_repository(repository_ctx)
        File "C:/windows/system32/tensorflow/third_party/gpus/cuda_configure.bzl", line 995, in _create_local_cuda_repository
                _get_cuda_config(repository_ctx)
        File "C:/windows/system32/tensorflow/third_party/gpus/cuda_configure.bzl", line 750, in _get_cuda_config
                _cudnn_version(repository_ctx, cudnn_install_base..., ...)
        File "C:/windows/system32/tensorflow/third_party/gpus/cuda_configure.bzl", line 466, in _cudnn_version
                find_cuda_define(repository_ctx, cudnn_header_dir, "c...", ...)
        File "C:/windows/system32/tensorflow/third_party/gpus/cuda_configure.bzl", line 422, in find_cuda_define
                auto_configure_fail(("Error reading %s: %s" % (str(h...)))
        File "C:/windows/system32/tensorflow/third_party/gpus/cuda_configure.bzl", line 210, in auto_configure_fail
                fail(("\n%sCuda Configuration Error:%...)))

Cuda Configuration Error: Error reading C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v9.1/include/cudnn.h: java.io.IOException: ERROR: src/main/native/windows/processes-jni.cc(239): CreateProcessW("grep" --color=never -A1 -E "#define CUDNN_MAJOR" "C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v9.1/include/cudnn.h"): Le fichier spÚcifiÚ est introuvable.

WARNING: Target pattern parsing failed.
ERROR: error loading package 'tensorflow/tools/pip_package': Encountered error while reading extension file 'cuda/build_defs.bzl': no such package '@local_config_cuda//cuda': Traceback (most recent call last):
        File "C:/windows/system32/tensorflow/third_party/gpus/cuda_configure.bzl", line 1142
                _create_local_cuda_repository(repository_ctx)
        File "C:/windows/system32/tensorflow/third_party/gpus/cuda_configure.bzl", line 995, in _create_local_cuda_repository
                _get_cuda_config(repository_ctx)
        File "C:/windows/system32/tensorflow/third_party/gpus/cuda_configure.bzl", line 750, in _get_cuda_config
                _cudnn_version(repository_ctx, cudnn_install_base..., ...)
        File "C:/windows/system32/tensorflow/third_party/gpus/cuda_configure.bzl", line 466, in _cudnn_version
                find_cuda_define(repository_ctx, cudnn_header_dir, "c...", ...)
        File "C:/windows/system32/tensorflow/third_party/gpus/cuda_configure.bzl", line 422, in find_cuda_define
                auto_configure_fail(("Error reading %s: %s" % (str(h...)))
        File "C:/windows/system32/tensorflow/third_party/gpus/cuda_configure.bzl", line 210, in auto_configure_fail
                fail(("\n%sCuda Configuration Error:%...)))

Cuda Configuration Error: Error reading C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v9.1/include/cudnn.h: java.io.IOException: ERROR: src/main/native/windows/processes-jni.cc(239): CreateProcessW("grep" --color=never -A1 -E "#define CUDNN_MAJOR" "C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v9.1/include/cudnn.h"): Le fichier spÚcifiÚ est introuvable.

INFO: Elapsed time: 0,544s

**Comment by FB43** [comment]
@ravikyram hello, I think the 1.8 tensorflow-gpu is correctly buid and installed with Python 2.7.
as you can see:
Name: tensorflow-gpu
Version: 1.8.0
Summary: TensorFlow helps the tensors flow
Home-page: https://www.tensorflow.org/
Author: Google Inc.
Author-email: opensource@google.com
License: Apache 2.0
Location: c:\python27\lib\site-packages
Requires: astor, tensorboard, numpy, termcolor, six, backports.weakref, absl-py, wheel, gast, protobuf, enum34, mock, grpcio

Now, I have a problem in the first Python script which is testing it:
Code:
from __future__ import absolute_import, division, print_function, unicode_literals
import tensorflow as tf
print("TF_Version {}".format(tfSession.__version__))
print("Keras_Version {}".format(tf.keras.__version__))
Result:
tf_version_check.py", line 5, in <module>
    print("TF_Version {}".format(tfSession.__version__))
NameError: name 'tfSession' is not defined
May you help me?
Thanks for all.
Best regards,
François

**Comment by FB43** [comment]
@angerson Hello, things are going a little bit more but still not succesful, now I have what you can see below. Forgot the previous messages, I've solved it. Thanks for your help. François
P:\Anaconda3\tensorflow>bazel build --config=v1 //tensorflow/tools/pip_package:b
uild_pip_package
WARNING: The following rc files are no longer being read, please transfer their
contents or import their path into one of the standard rc files:
p:\anaconda3\tensorflow/tools/bazel.rc
WARNING: Ignoring JAVA_HOME, because it must point to a JDK, not a JRE.
INFO: Options provided by the client:
  Inherited 'common' options: --isatty=0 --terminal_columns=80
INFO: Options provided by the client:
  'build' options: --python_path=p:/Anaconda3/python.exe
INFO: Reading rc options for 'build' from p:\anaconda3\tensorflow\bazel.rc:
  'build' options: --define framework_shared_object=true --define=use_fast_cpp_p
rotos=true --define=allow_oversize_protos=true --define=grpc_no_ares=true --spaw
n_strategy=standalone --genrule_strategy=standalone -c opt
INFO: Reading rc options for 'build' from p:\anaconda3\tensorflow\.tf_configure.
bazelrc:
  'build' options: --action_env PYTHON_BIN_PATH=p:/Anaconda3/python.exe --action
_env PYTHON_LIB_PATH=p:/Anaconda3/lib/site-packages --force_python=py3 --host_fo
rce_python=py3 --python_path=p:/Anaconda3/python.exe --define with_xla_support=t
rue --define with_gdr_support=true --define with_verbs_support=true --action_env
 TF_NEED_OPENCL_SYCL=0 --action_env TF_NEED_CUDA=1 --action_env CUDA_TOOLKIT_PAT
H=C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v9.1 --action_env TF_CUDA_V
ERSION=9.1 --action_env CUDNN_INSTALL_PATH=C:/Program Files/NVIDIA GPU Computing
 Toolkit/CUDA/v9.1 --action_env TF_CUDNN_VERSION=9 --action_env TF_CUDA_COMPUTE_
CAPABILITIES=3.0 --action_env TF_CUDA_CLANG=0 --action_env CUDA_PATH=C:/Program
Files/NVIDIA GPU Computing Toolkit/CUDA/v9.1 --action_env CUDA_COMPUTE_CAPABILIT
IE=None --action_env NO_WHOLE_ARCHIVE_OPTION=1 --config=win-cuda --define grpc_n
o_ares=true --copt=-DGEMMLOWP_ALLOW_SLOW_SCALAR_FALLBACK --host_copt=-DGEMMLOWP_
ALLOW_SLOW_SCALAR_FALLBACK --config monolithic --copt=-w --host_copt=-w --verbos
e_failures
INFO: Found applicable config definition build:win-cuda in file p:\anaconda3\ten
sorflow\bazel.rc: --define=using_cuda=true --define=using_cuda_nvcc=true
INFO: Found applicable config definition build:monolithic in file p:\anaconda3\t
ensorflow\bazel.rc: --define framework_shared_object=false
ERROR: Config value v1 is not defined in any .rc file

P:\Anaconda3\tensorflow>

**Comment by mohantym** [comment]
Hi @FB43 ,  Could you please look at this[ link ](https://www.tensorflow.org/install/source_windows#tensorflow_1x)for installing from source in windows.
We also see that you are using old version of Tensorflow which is officially considered as end of life, We recommend that you upgrade to 2.6 version and let us know if the issue still persists in newer versions .Please open a new issue in case you face any errors, we will get you the right help .Thanks!
